static/app/utils/platform.tsx

```diff
@@ -1,4 +1,32 @@
-import {mobile} from 'sentry/data/platformCategories';
+import {
+  backend,
+  desktop,
+  frontend,
+  mobile,
+  PlatformCategory,
+  serverless,
+} from 'sentry/data/platformCategories';
+
+const platformCategoryMap = {};
+mobile.forEach(platform => (platformCategoryMap[platform] = PlatformCategory.mobile));
+frontend.forEach(platform => (platformCategoryMap[platform] = PlatformCategory.frontend));
+backend.forEach(platform => (platformCategoryMap[platform] = PlatformCategory.backend));
+serverless.forEach(
+  platform => (platformCategoryMap[platform] = PlatformCategory.backend)
+);
+desktop.forEach(platform => (platformCategoryMap[platform] = PlatformCategory.backend));
+
+/**
+ *
+ * @param platform - a SDK platform, for example `node-express`, `javascript-react`
+ * @returns - the platform category, for example `backend`, `serverless`
+ */
+export function platformToCategory(platform: string | undefined): PlatformCategory {
+  if (!platform || !platformCategoryMap[platform]) {
+    return PlatformCategory.other;
+  }
+  return platformCategoryMap[platform];
+}
```

Comment:
This works, but for the sake of readability, how about we just use the function to logically convert the platform to the enum instead of creating a map?

Something like this for example:

```
function platformToCategory( ... ) {
	if (mobile.includes(platform)) {
		return PlatformCategory.MOBILE
	}

	if (frontend.includes(platform)) {
		return PlatformCategory.FRONTEND
	}

	...etc
}
```

Reply:
This is actually how it originally was and was because mobile/frontend arrays are these readonly constants requiring you to do some funky string array conversion like this https://github.com/getsentry/sentry/blob/35d73d9291e36806708e98387cf68c6c8af517af/static/app/utils/platform.tsx#L49

After previous code reviews regarding that, this is what I did as an alternative

Reply:
lol honestly it seems like the root of the problem here is that these `mobile`, `frontend`, etc arrays are initialized as const string arrays when they should've just been enums to begin with, but it's going to be annoying to change that now. 

With how it's setup I personally don't find much issue with the original method via `return ([...mobile] as string[]).includes(platform);`. I'm not a huge fan of the map construction method since it feels a bit unnecessarily verbose but I'll leave the final decision up to you

Reply:
Tbh, I think the original method was a bit more readible, but I believe the map construction would be more performant (hash map vs array lookup), definitely not a concern for 100 items. I'll change it back!

Reply:
Yep, it's constant time in both cases because we have a set number of elements

Reply:
made the change!